### PR TITLE
update to ocaml-dns latest release

### PR DESCRIPTION
**Pull request by palainp**
With the lastest releases of ocaml-dns I failed to compile the unikernel. With the modifications in this PR it compiles and runs fine as previously with an up to date Qubes 4.0 (I did not tryied with the 4.1 rc1).

### CODE REVIEW

**Comment by hannesm** [comment]
Thanks for your updates. I added some commits and updated the most recent checksum of a docker build. The build output is available at http://data.robur.coop/mirage-firewall.tar.bz2

//cc @talex5 would you mind to review and merge if you're happy?

**Comment by hannesm** [comment]
sorry, force-pushed to include a commit that reads the DNS resolver from qubesDB (instead of hardcoding a custom resolver). this compiles fine, the binary at  http://data.robur.coop/mirage-firewall.tar.bz2 has been updated. if anyone could check that it works nicely (especially with firewall rules that require DNS resolution), that would be great :)

File: my_dns.ml [hunk 1]
```diff
@@ -3,22 +3,22 @@ open Lwt.Infix
...
   type context = { t : t ; timeout_ns : int64 ref; mutable src_port : int }
 
-  let nameserver t = t.nameserver
+  let nameservers t = t.nameservers
   let rng = R.generate ?g:None
   let clock = C.elapsed_ns
 
-  let create ?(nameserver = `UDP, (Ipaddr.V4.of_string_exn "91.239.100.100", 53)) ~timeout stack =
-    { nameserver ; stack ; timeout_ns = timeout }
+  let create ?(nameservers = `Udp, [(Ipaddr.V4.of_string_exn "91.239.100.100", 53)]) ~timeout stack =
```
Comment (talex5) [inline my_dns.ml:20]: Not part of this PR, but what is this magic IP address doing here? Should this be a configuration setting?

**Review by talex5 — COMMENTED** [review]
I'm not running Qubes at the moment, so I haven't tested this, but it looks reasonable. It's a shame we don't have CI any more (travis-ci.org has shut down and the replacement wants payment details). I guess we should either move to GHA, or wait for Mirage 4 and use ocaml-ci.

File: my_dns.ml [hunk 2]
```diff
@@ -3,22 +3,22 @@ open Lwt.Infix
 module Transport (R : Mirage_random.S) (C : Mirage_clock.MCLOCK) = struct
   type +'a io = 'a Lwt.t
   type io_addr = Ipaddr.V4.t * int
-  type ns_addr = [ `TCP | `UDP ] * io_addr
+  type ns_addr = Dns.proto * io_addr list
```
[the remark this answers is not in the document]
Comment (hannesm) [reply my_dns.ml]: I addressed this in c4f9142. The DNS client in qubes-mirage-firewall uses a single DNS server, from QubesDB `/qubes-primary-dns`.

**Comment by hannesm** [comment]
FWIW, the QubesDB key documentation (including /qubes-primary-dns) is at https://www.qubes-os.org/doc/vm-interface/

**Comment by talex5** [comment]
OK, thanks!
